src/middle/SizedType.ml

```diff
@@ -229,6 +238,16 @@ let rec has_mem_pattern = function
   | SArray (t, _) -> has_mem_pattern t
   | STuple subtypes -> List.exists ~f:has_mem_pattern subtypes
 
+let is_eigen_type st =
+  match st with
+  | (SVector (mem, _) | SRowVector (mem, _) | SMatrix (mem, _, _))
+    when Mem_pattern.is_opencl mem ->
+      false
+  | SVector _ | SRowVector _ | SMatrix _ | SComplexRowVector _
+   |SComplexVector _ | SComplexMatrix _ ->
+      true
+  | _ -> false
+
```

Comment:
I feel a little nervous that `SizedType.is_eigen_type st` and `UnsizedType.is_eigen_type (SizedType.to_unsized st)` would return different things in some circumstances. I suppose it depends on how/where both of them are used, but I'd feel a bit more confident with a `is_eigen_type` which matches Unsized and using `is_eigen st && not (is_opencl st)` where needed.

What do you think?

Reply:
Yeah I agree it's very shaky. I think splitting it out it a better idea